lib/Console/Application.php

```diff
@@ -22,22 +22,41 @@
 
 use \Symfony\Component\Console\Input\InputInterface;
 use \Symfony\Component\Console\Output\OutputInterface;
+use \Symfony\Component\Console\ConsoleEvents;
+use \Symfony\Component\EventDispatcher\EventDispatcher;
+use \Symfony\Component\Console\Event\ConsoleCommandEvent;
 
 class Application extends \Symfony\Component\Console\Application
 {
     public function __construct()
     {
         parent::__construct('Mesamatrix CLI', \Mesamatrix::$config->getValue('info', 'version'));
+
+        $dispatcher = new EventDispatcher();
+
+        // Install CLI logger
+        $dispatcher->addListener(ConsoleEvents::COMMAND,
+            function (ConsoleCommandEvent $e)
+            {
+                $logger = new \Monolog\Logger('cli.'.$e->getCommand()->getName());
+                foreach (\Mesamatrix::$logger->getHandlers() as $handler) {
+                    $logger->pushHandler($handler);
+                }
+                $logger->pushHandler(
+                    new \Symfony\Bridge\Monolog\Handler\ConsoleHandler($e->getOutput())
+                );
+                \Mesamatrix::$logger = $logger;
```

Comment:
Yes, unfortunately it seems the only way to get a different name for the logger is to create a completely new one. IMHO, it is important to see in the logs which logger wrote that log entry, be it from the CLI or from the web interface. At the same time, I thought it would be neat to have the command name in the log, which is why I implemented the EventDispatcher thing.